tstest/integration/integration_test.go

```diff
@@ -1529,3 +1522,140 @@ func TestEncryptStateMigration(t *testing.T) {
 		runNode(t, wantPlaintextStateKeys)
 	})
 }
+
+func TestC2NDebugNetmap(t *testing.T) {
+	tstest.Shard(t)
+	tstest.Parallel(t)
+	env := NewTestEnv(t)
+
+	var testNodes []*TestNode
+	var nodes []*tailcfg.Node
+	for i := range 2 {
+		n := NewTestNode(t, env)
+		d := n.StartDaemon()
+		defer d.MustCleanShutdown(t)
+
+		n.AwaitResponding()
+		n.MustUp()
+		n.AwaitRunning()
+		testNodes = append(testNodes, n)
+
+		controlNodes := env.Control.AllNodes()
+		if len(controlNodes) != i+1 {
+			t.Fatalf("expected %d nodes, got %d nodes", i+1, len(controlNodes))
+		}
+		for _, cn := range controlNodes {
+			// Make sure `nodes` are in the same order as `testNodes`.
+			if !slices.ContainsFunc(nodes, func(n *tailcfg.Node) bool { return n.Key == cn.Key }) {
+				nodes = append(nodes, cn)
+			}
+		}
+	}
+
+	// getNetmap fetches the current netmap. If a candidate map response is provided,
+	// a candidate netmap is also fetched and compared to the current netmap.
```

Comment:
```suggestion
	// getNetmap fetches the current netmap through the C2N /debug/netmap endpoint.
    // If a candidate map response is provided, it is sent to the node and the returned
    // Current/Candidate netmaps are compared.
```

Maybe rename to getC2NNetmap?